ui/org.eclipse.pde.core/src/org/eclipse/pde/internal/core/target/P2TargetUtils.java

```diff
@@ -1078,11 +1078,24 @@ private void resolveWithPlanner(ITargetDefinition target, IProgressMonitor monit
 			request.setInstallableUnitProfileProperty(unit, PROP_INSTALLED_IU, Boolean.toString(true));
 		}
 
-		ProvisioningContext context = new ProvisioningContext(getAgent());
+		List<IArtifactRepository> extraArtifactRepositories = new ArrayList<>();
+		List<IMetadataRepository> extraMetadataRepositories = new ArrayList<>();
+		getAdditionalProvisionIUs(target, extraArtifactRepositories, extraMetadataRepositories);
+		ProvisioningContext context = new ProvisioningContext(getAgent()) {
+			@Override
+			public IQueryable<IArtifactRepository> getArtifactRepositories(IProgressMonitor monitor) {
+				return QueryUtil.compoundQueryable(super.getArtifactRepositories(monitor),
+						(query, ignore) -> query.perform(extraArtifactRepositories.iterator()));
```

Comment:
```suggestion
						QueryUtil.compoundQueryable(extraArtifactRepositories));
```
Can't we use QueryUtil.compoundQueryable() for a list of artifact repositories, like it is done for metadata-repositories?